light/client.go

```diff
@@ -752,8 +754,18 @@ func (c *Client) bisection(
 				return fmt.Errorf("verify non adjacent from #%d to #%d failed: %w",
 					trustedHeader.Height, headerCache[depth].sh.Height, err)
 			}
+			newProviderHeader, newProviderVals, err := c.fetchHeaderAndValsAtHeight(newHeader.Height)
```

Comment:
I think we should just decline to change provider during bisection. If primary fails to provide header during bisection, we start from latest saved state and new primary

Reply:
yeah this is what it's doing, replacing primary and then restarting the entire bisection algorithm instead of just continuing or do you mean we should return the error back and let the handler of the light client remove the primary and try again if they want

Reply:
> replacing primary and then restarting the entire bisection algorithm instead of just continuing or do you mean we should return the error back and let the handler of the light client remove the primary and try again if they want

the former. thanks